### PR TITLE
Direct to all in one page for search

### PR DESCRIPTION
**Pull request by ErinBecker**
The goal of this PR (in combination with SANDPAPER PR LINK) is to:
- [x] Remove the inactive search box icon from lesson pages (this PR)
- [x] Replace with a button pointing towards the lesson's all in one page (this PR)
- [x] Make search button title translatable (https://github.com/ErinBecker/sandpaper/tree/translate-aio)
- [ ] Add a banner to the top of the all in one page directing folks to use ctl-f for search (SANDPAPER PR LINK) 

The screenshot below shows the replacement of the search box icon with a button pointing towards the lesson's all in one page: 
Screenshot 2024-03-22 at 3 13 34 PM

Note to self and @froggleston - if we end up merging this, please squash and merge, there's a lot of commits where I was experimenting.

Fixes https://github.com/carpentries/workbench/issues/8

### CODE REVIEW

File: inst/pkgdown/templates/header.html [hunk 1]
```diff
@@ -189,6 +190,8 @@
         </button>
       </fieldset>
     </form>
+    -->
+    <a class="btn btn-primary" href="{{#site}}{{root}}{{/site}}aio.html" role="button">Search in this lesson's "all in one" page</a>
```
Comment (ErinBecker) [inline inst/pkgdown/templates/header.html:194]:
```suggestion
    <a class="btn btn-primary" href="{{#site}}{{root}}{{/site}}aio.html" role="button">{{ translate.SearchButton }}</a>
```

**Comment by ErinBecker** [comment]
With https://github.com/ErinBecker/sandpaper/tree/translate-aio, allows for translation of the new "Search the All In One page" button, however, have discovered a problem - need to not render this button on workshop overview pages, as they don't have an All In One page. 

Note to self, look at the code linked below to figure out how to make addition of page elements dependent on whether a lesson is an overview page:
- https://github.com/carpentries/sandpaper/blob/3b416710d35ae1a18c8b55e4c2227c296a91fe99/R/build_site.R#L61
- https://github.com/carpentries/sandpaper/blob/3b416710d35ae1a18c8b55e4c2227c296a91fe99/R/build_site.R#L21C7-L21C64

**Comment by ErinBecker** [comment]
This now adds the search all in one button to the top of any lesson that is not a workshop overview (Thanks @froggleston!), and the search button is translatable through the .po files.

**Comment by froggleston** [comment]
LGTM! Awesome work! 🚀

**Pull request merged**